src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs

```diff
@@ -9,42 +14,232 @@ private FileVersionInfo(string fileName)
+            return false;
+        }
+
+        /// <summary>Load our fields from the metadata of the file as represented by the provided metadata reader.</summary>
+        /// <param name="metadataReader">The metadata reader for the CLI file this represents.</param>
+        private void LoadManagedAssemblyMetadata(MetadataReader metadataReader)
+        {
+            AssemblyDefinition assemblyDefinition = metadataReader.GetAssemblyDefinition();
+
+            // Set the internal and original names based on the file name.
+            _internalName = _originalFilename = Path.GetFileName(_fileName);
+
+            // Set the product version based on the assembly's version (this may be overwritten 
+            // later in the method).
+            Version productVersion = assemblyDefinition.Version;
+            _productVersion = productVersion.ToString(4);
+            _productMajor = productVersion.Major;
+            _productMinor = productVersion.Minor;
+            _productBuild = productVersion.Build;
+            _productPrivate = productVersion.Revision;
+
+            // "Language Neutral" is used on Win32 for unknown language identifiers.
+            _language = "Language Neutral";
 
+            // Set other fields to default values in case they're not overwritten by attributes
             _companyName = string.Empty;
-            _fileDescription = string.Empty;
-            _fileVersion = string.Empty;
-            _internalName = string.Empty;
-            _legalCopyright = string.Empty;
-            _originalFilename = string.Empty;
-            _productName = string.Empty;
-            _productVersion = string.Empty;
             _comments = string.Empty;
+            _fileDescription = " "; // this is what the managed compiler outputs when value isn't set
+            _fileVersion = string.Empty;
+            _legalCopyright = " "; // this is what the managed compiler outputs when value isn't set
             _legalTrademarks = string.Empty;
+            _productName = string.Empty;
             _privateBuild = string.Empty;
             _specialBuild = string.Empty;
 
-            _language = string.Empty;
-
-            _fileMajor = 0;
-            _fileMinor = 0;
-            _fileBuild = 0;
-            _filePrivate = 0;
-            _productMajor = 0;
-            _productMinor = 0;
-            _productBuild = 0;
-            _productPrivate = 0;
-
+            // Be explicit about initialization to suppress warning about fields not being set
             _isDebug = false;
             _isPatched = false;
-            _isPrivateBuild = false;
             _isPreRelease = false;
+            _isPrivateBuild = false;
             _isSpecialBuild = false;
+
+            // Everything else is parsed from assembly attributes
+            MetadataStringComparer comparer = metadataReader.StringComparer;
+            foreach (CustomAttributeHandle attrHandle in metadataReader.GetAssemblyDefinition().GetCustomAttributes())
```

Comment:
This can throw InvalidOperationException in GetAssemblyDefinition() if you've opened a .netmodule. There is an IsAssembly to bail prior to that.